TU20Bot/Commands/NameMatch.cs

```diff
@@ -0,0 +1,154 @@
+                     * If both of them match then give the user the role passed in the command
+                     * If first name doesn't match then notify in the console that the last name 
+                     * match was successful.
+                     * If last name doesn't match then notify the user in the console. 
+                     */
+                    else {
+                        string firstName = fullName.Substring(0, spaceIndex);
+                        string lastName = fullName.Substring(spaceIndex + 1);
+
+                        // Checking if the last names match
+                        if (compare(origNames[i, 1], lastName) ||
+                            compare(lastName, origNames[i, 1])) {
+                            // Checking if the first names match
+                            if (compare(origNames[i, 0], firstName) ||
+                                compare(firstName, origNames[i, 0])) {
+                                // Assigning user a role specified in the command
+                                var role = Context.Guild.GetRole(roleId);
+                                await (user as IGuildUser).AddRoleAsync(role);
+                                discordMessage +=
+                                    $"`{firstName} {lastName}` has been granted role `{role}`\n";
+                                break;
+                            }
+
+                            // If first name doesn't match
+                            noLastNameMsg += $"`{lastName}` of `{firstName} {lastName}` matches " +
+                                             $"with `{origNames[i, 1]}` of `{origNames[i, 0]} {origNames[i, 1]}`\n";
+                            Console.WriteLine($"`{lastName}` of `{firstName} {lastName}` matches " +
+                                              $"with `{origNames[i, 1]}` of `{origNames[i, 0]} {origNames[i, 1]}`");
+                        }
+                    }
+                }
+            }
+
+            // For adding all the messages to send to the user
+            discordMessage += noLastNameMsg + noFirstNameMsg;
+            // For reporting if the user doesn't exist in the server (no last name match)
+            for (int i = 0; i < origNames.GetLength(0); i++) {
+                if (!discordMessage.Contains(origNames[i, 1])) {
+                    notInGuildMsg +=
+                        $"{origNames[i, 0]} {origNames[i, 1]} did not have a last name match therefore is not in the server\n";
+                }
+            }
+
+            discordMessage += notInGuildMsg;
+            await sendSplitMessage(discordMessage, "\n");
+        }
+
+        private bool compare(string name, string matchName) {
+            return name.Replace(" ", "").Contains(matchName.Replace(" ", ""));
+        }
+
+        private async Task sendSplitMessage(string mainMessage, string separator) {
+            // The max limit set by discord
+            int messageLimit = 2000;
```

Comment:
If this won't be changed, it should be `readonly` and defined at the beginning of the class.